cmd/compare.go

```diff
@@ -66,3 +82,137 @@ func rebuildCompareCommand() {
+	if err != nil {
+		logger.Fatal().Err(err).Msgf("Failed to find rule file for rule id %s", ruleId)
+	}
+	if matches == nil || len(matches) > 1 {
+		logger.Fatal().Msgf("Failed to find rule file for rule id %s", ruleId)
+	}
+
+	filePath := matches[0]
+	logger.Debug().Msgf("Processing data file %s", filePath)
+
+	currentRegex := readCurrentRegex(filePath, ruleId, chainOffset)
+	compareRegex(filePath, ruleId, chainOffset, regex, currentRegex)
+}
+
+func readCurrentRegex(filePath string, ruleId string, chainOffset uint8) string {
+	contents, err := os.ReadFile(filePath)
+	if err != nil {
+		logger.Fatal().Err(err).Msgf("Failed to read rule file %s", filePath)
+	}
+
+	lines := bytes.Split(contents, []byte("\n"))
+
+	idRegex := regexp.MustCompile(fmt.Sprintf("id:%s", ruleId))
+	index := 0
+	var line []byte
+	for index, line = range lines {
+		if idRegex.Match(line) {
+			break
+		}
+	}
+	regexLine := lines[index-1]
+	regexRegex := regexp.MustCompile(`.*"@rx (.*)" \\`)
+	found := regexRegex.FindAllStringSubmatch(string(regexLine), -1)
+	if len(found) == 0 {
+		logger.Fatal().Msgf("Failed to find rule %s in %s", ruleId, filePath)
+	}
+	return found[0][1]
+}
+
+func compareRegex(filePath string, ruleId string, chainOffset uint8, generatedRegex string, currentRegex string) {
+	if currentRegex == generatedRegex {
+		fmt.Println("Regex of ", ruleId, " has not changed")
+		return
+	}
+
+	fmt.Println("Regex of", ruleId, " has changed!")
+	diffFound := false
+	maxChunks := int(math.Ceil((math.Max(float64(len(currentRegex)), float64(len(generatedRegex))) / 50)))
+	for index := 0; index < maxChunks*50; index += 50 {
+		currentChunk := ""
+		generatedChunk := ""
+		counter := ""
+		endIndex := int(math.Min(float64(len(currentRegex)), float64(index+50)))
+		if endIndex > index {
+			currentChunk = currentRegex[index:endIndex]
+		}
+		endIndex = int(math.Min(float64(len(generatedRegex)), float64(index+50)))
+		if endIndex > index {
+			generatedChunk = generatedRegex[index:endIndex]
+		}
+
+		printFirstDiff := !diffFound && currentChunk != generatedChunk
+
+		if printFirstDiff {
+			diffFound = true
+			fmt.Printf("\n===========\nfirst difference\n-----------")
+		}
+		if currentChunk != "" {
+			fmt.Printf("\ncurrent:  ")
+			fmt.Print(strings.Repeat(" ", 5), currentChunk)
+			counter := fmt.Sprint("(", (index/50)+1, " / ", maxChunks, ")")
+			if currentChunk != generatedChunk {
+				counter = "~ " + counter
+			}
+			fmt.Print(strings.Repeat(" ", 60-len(currentChunk)), counter)
+		}
+		if generatedChunk != "" {
+			fmt.Printf("\ngenerated: ")
+			fmt.Print(strings.Repeat(" ", 4), generatedChunk)
+			counter = fmt.Sprint("(", (index/50)+1, " / ", maxChunks, ")")
+			if currentChunk != generatedChunk {
+				counter = "~ " + counter
+			}
+			fmt.Println(strings.Repeat(" ", 59-len(generatedChunk)), counter)
+		}
+		if printFirstDiff {
+			fmt.Println("===========")
+		}
+	}
+	fmt.Printf("\n")
+}
```

Comment:
Why not https://github.com/terrastruct/diff or similar?

Reply:
Hm... Let me look into that.

Reply:
NO!
I TALKA WITTA MY HANDSA!

Reply:
Ah, then I SHATAPA WITH MY SHATAPA.